### DISCUSSION TITLE
About Halo infinite Custom touch controls

### DISCUSSION THREAD

**Discussion by Tak-attack** [redphx/better-xcloud#265] [Custom touch controls]
I have created a custom layout for Halo infinite.
It incorporates gyro and touchpad functionality, and so far (for me) it is a satisfactory custom layout, but I have concerns about two things.

![Screenshot_2024-02-22-04-06-15-299_com kiwibrowser browser](https://github.com/redphx/better-xcloud/assets/149154024/87c3f9e0-619f-4752-ad5d-3bf08900e64a)

![Screenshot_2024-02-22-04-05-53-285_com kiwibrowser browser](https://github.com/redphx/better-xcloud/assets/149154024/b0f011a7-0720-4e1e-9945-ab39738d124f)

1. halo infinite is a single campaign game, but it is also a multiplayer online game. Using an unofficial custom layout in an online game could be considered cheating, but is it acceptable to introduce a custom layout in such a game as well?

2. We also have another layout that replaces the "relativeMouse" function for players who do not like to use stick-based aim. However, after testing it in my environment, I found that it works correctly for console remote play, but not for cloud play with relativeMouse.
I annotated the name of that layout with (Remote play only), but can I include a layout that does not work in cloud play?

Let me know your various thoughts. Thanks for taking a look.

**Comment by redphx** [comment]
> is it acceptable to introduce a custom layout in such a game as well?

Yes, that's not cheating.

> I found that it works correctly for console remote play, but not for cloud play with relativeMouse.

Maybe it's a bug. It should work in both cases.

**Reply by Tak-attack** [reply]
Thank you very much. I will make a pull request for a custom layout later.
I will remove the "Remote play only" part of the name of the relativeMouse mode.
